scalding-core/src/main/scala/com/twitter/scalding/typed/cascading_backend/CascadingBackend.scala

```diff
@@ -100,10 +103,28 @@ object CascadingBackend {
         op(ts, keyF)
     }
 
-   private case class CascadingPipe[T](pipe: Pipe,
+   private case class CascadingPipe[+T](pipe: Pipe,
     fields: Fields,
     @transient localFlowDef: FlowDef, // not serializable.
-    converter: TupleConverter[T])
+    converter: TupleConverter[_ <: T]) {
+
+    /**
+     * merge the flowDef into this new flowdef an make sure the tuples
+     * have the structure defined by setter
+     */
+    def toPipe[U >: T](f: Fields, fd: FlowDef, setter: TupleSetter[U]): Pipe = {
+      // TODO, this may be identity if the setter is the inverse of the
+      // converter. If we can identify this we will save allocations
+      val resFd = new RichFlowDef(fd)
+      resFd.mergeFrom(localFlowDef)
```

Comment:
merge from doesn't mutate `localFlowDef` i imagine?

Reply:
that's right. We have been using that for a while. It mutates the left, but not the right.

Reply:
👍